metabot.py

```diff
@@ -1,25 +1,53 @@
-# import smbus
-import time, sys, subprocess, math, threading
-import socket, pprint, struct, json
 import getopt
+import json
+import logging
+import math
+import pprint
 import serial
+import socket
+import struct
+import subprocess
+import sys
+import threading
+import time
+
 from ast import literal_eval as make_tuple
 
+_log = logging.getLogger("metabot")
+# Setup file logging
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(name)s:%(asctime)s:%(levelname)s:"
+           "%(funcName)s(%(lineno)d): %(message)s",
+    filename="./metabot.log"
+)
+# Setup screen logging
+console = logging.StreamHandler()
+console.setLevel(logging.INFO)
+formatter = logging.Formatter('%(name)-12s: %(levelname)-8s %(message)s')
+console.setFormatter(formatter)
+logging.getLogger('').addHandler(console)
+
+_log.info("Started up")
+
+ipaddress = "10.5.5.1"
+
 
-class serial_port():
+class SerialPort:

```

Comment:
In Python 2, classes should inherit from `object`:

```python
class SerialPort(object):
```

This is the distinction between "old-style classes" (which don't inherit from object) and "new-style classes" (which do). The new-style classes tidy up a lot of cruft around types/classes, and are the recommended approach.

In Python 3, there are only new-style classes, but subclassing from object is still recommended.